### DISCUSSION TITLE
Missing data for comonth > flodfreqcl

### DISCUSSION THREAD

**Discussion by kevinwolz** [ncss-tech/soilDB#331] [Q&A] [answered]
There are a handful of surveys across the country where comonth > flodfreqcl is not defined. This occurs even if other soils of the same musym do have flodfreqcl defined. For example, here are two mukeys of the same musym:
![image](https://github.com/ncss-tech/soilDB/assets/11711566/fee7588b-0070-41ea-86bc-250698b24078)

```
SELECT co.mukey, co.cokey, co.comppct_r, com.monthseq, com.flodfreqcl, com.floddurcl
FROM component AS co 
LEFT JOIN comonth AS com ON com.cokey = co.cokey
WHERE mukey IN ('606799', '178908') ORDER BY co.mukey
```

![image](https://github.com/ncss-tech/soilDB/assets/11711566/085ab83e-c00e-4b22-8905-44432def86fb)

You can see in ESRI's compiled map of flooding frequency all of the missing data. For example, West Virginia is particularly bad:
![image](https://github.com/ncss-tech/soilDB/assets/11711566/01bf6d86-9048-4bd5-9ad5-f2cd4cc12208)

Short of contacting every soil scientist in every survey area that has missing data, what might you suggest as a way to "hack" the missing data for these soils. For example, is it reasonable to say that, if flodfreqcl is missing for a mukey, one can fill it in with data from a different mukey with the same musym? If even if that is missing, could one look for other mukeys with the same natmusym?

**Comment by brownag** [comment, marked as answer]
> Short of contacting every soil scientist in every survey area that has missing data, what might you suggest as a way to "hack" the missing data for these soils. 

I think you can probably assume that the records that you are creating with the LEFT JOIN that contain `NA` have a flodfreqcl of None for all months. This is not how it should be populated, but in general if it does flood it should be in there.

> For example, is it reasonable to say that, if flodfreqcl is missing for a mukey, one can fill it in with data from a different mukey with the same musym? 

No, in general musym should not be used to identify similar concepts. It is very common that musyms for adjacent survey areas do not correspond to the exact same concept--especially when _numeric_ musyms were applied to the alphabetized list of mapunits from that particular survey area--but in general we can't rely on the published survey mapunit symbol to relate similar concepts from different areas

> If even if that is missing, could one look for other mukeys with the same natmusym?

Mapunits that share the same national musym have the same data, so this would not solve your problem.

**Reply by kevinwolz** [reply]
Thanks! We will fill in the missing data with "None".
